File: HearstMappingsEditor/Scripts/app/references/source/dimProject.js
$(function () {
    $(".dim-project-ref-tab").addClass("active");
});

function getFilterData(skip, take) {
    var projectGroups = [];
    var printDigitals = [];

    $.each($(".filters .project-group").select2("data"), function (index, item) {
        var id = item.id;
        if (id == notSelected) {
            id = "";
        }
        projectGroups.push(id);
    });
    $.each($(".filters .print-digital").select2("data"), function (index, item) {
        printDigitals.push(item.id);
    });

    return {
        ProjectCode: $(".filters .project-code").val(),
        ProjectGroups: projectGroups,
        ManagementProject: $(".filters .management-project").val(),
        ManagementParent: $(".filters .management-parent").val(),
        ManagementBrand: $(".filters .management-brand").val(),
        PrintDigitals: printDigitals,
        Type: $(".filters .type").val(),
        Description: $(".filters .description").val(),
        C1HypCode: $(".filters .c1-hyp-code").val(),
        C2HypCodeNew: $(".filters .c2-hyp-code-new").val(),
        C2Management: $(".filters .c2-management").val(),
        SortMode: getSortMode(),
        Skip: skip,
        Take: take == null ? pageSize : take
    };
}

function getNewItemDataJson() {
    return JSON.stringify({
        item: {
            ProjectCode: $("#nm-project-code").val(),
            ProjectGroup: $("#nm-project-group").select2("data")[0].id,
            ManagementProject: $("#nm-management-project").val(),
            ManagementParent: $("#nm-management-parent").val(),
            ManagementBrand: $("#nm-management-brand").val(),
            PrintDigital: $("#nm-print-digital").select2("data")[0].id,
            Type: $("#nm-type").val(),
            Description: $("#nm-description").val(),
            C1HypCode: $("#nm-c1-hyp-code").val(),
            C2HypCodeNew: $("#nm-c2-hyp-code-new").val(),
            C2Management: $("#nm-c2-management").val()
        }
    });
}

function getAddedOrEdited() {
    var addedOrEdited = [];

    $(".main-table tbody tr.changed").each(function () {
        var row = $(this);
        addedOrEdited.push({
            ProjectID: $(".readonly-cell.project-id", row).text(),
            ProjectCode: $(".text-input-cell.project-code", row).data("val"),
            ProjectGroup: $(".readonly-cell.project-group", row).data("val"),
            ManagementProject: $(".text-input-cell.management-project", row).data("val"),
            ManagementParent: $(".text-input-cell.management-parent", row).data("val"),
            ManagementBrand: $(".text-input-cell.management-brand", row).data("val"),
            PrintDigital: $(".readonly-cell.print-digital", row).data("val"),
            Type: $(".text-input-cell.type", row).data("val"),
            Description: $(".text-input-cell.description", row).data("val"),
            C1HypCode: $(".text-input-cell.c1-hyp-code", row).data("val"),
            C2HypCodeNew: $(".text-input-cell.c2-hyp-code-new", row).data("val"),
            C2Management: $(".text-input-cell.c2-management", row).data("val")
        });
    });

    return addedOrEdited;
}

File: HearstMappingsEditor.Data.Models/Entities/References/Source/DimProject.cs
using HearstMappingsEditor.Common;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HearstMappingsEditor.Data.Models
{
	[Table("DimProject", Schema = "Source")]
    public class DimProject : BaseEntity, IReference
    {
        [ExcelExport(Order = 0)]
        [Key]
        public long ProjectID { get; set; }

        [ExcelExport(Order = 1)]
        [MaxLength(50)]
        public string ProjectCode { get; set; }

        [ExcelExport(Order = 2)]
        [MaxLength(50)]
        public string ProjectGroup { get; set; }

        [ExcelExport(Order = 3)]
        [MaxLength(50)]
        public string ManagementProject { get; set; }

        [ExcelExport(Order = 4)]
        [MaxLength(50)]
        public string ManagementParent { get; set; }

        [ExcelExport(Order = 5)]
        [MaxLength(50)]
        public string ManagementBrand { get; set; }

        [ExcelExport(Order = 6)]
        [Column("Print Digital")]
        [MaxLength(50)]
        public string PrintDigital { get; set; }

        [ExcelExport(Order = 7)]
        [MaxLength(50)]
        public string Type { get; set; }

        [ExcelExport(Order = 8)]
        [Column("Описание")]
        [MaxLength(50)]
        public string Description { get; set; }

        [ExcelExport(Order = 9)]
        [MaxLength(50)]
        public string C1HypCode { get; set; }

        [ExcelExport(Order = 10)]
        [MaxLength(50)]
        public string C2HypCodeNew { get; set; }

        [ExcelExport(Order = 11)]
        [MaxLength(50)]
        public string C2Management { get; set; }

        [ExcelExport(Order = 12)]
        public DateTime? CreateDate { get; set; }

        [NotMapped]
        public long Id => ProjectID;
    }
}
